src/biogeochem/CNPhenologyMod.F90

```diff
@@ -972,6 +1034,147 @@ subroutine CNSeasonDecidPhenology (num_soilp, filter_soilp       , &
+                       (critical_offset_high_lat-abs(grc%latdeg(g)))
+           if (crit_daylat < crit_dayl) then
+              crit_daylat = crit_dayl !maintain previous offset from White 2001 as minimum
+           end if
+        end if
+    ! Critical day length depends just on vegetation type
+    case(critical_daylight_depends_on_veg)
+        if ( pftcon%season_decid_temperate(patch%itype(p)) == 1 )then
+           crit_daylat = crit_dayl
+        else
+           crit_daylat=params_inst%crit_dayl_at_high_lat
+        end if
+    ! Critical day length depends on latitude
+    case(critical_daylight_depends_on_lat)
+        ! Critical daylength is higher at high latitudes and shorter
+        ! for temperatre regions
+        crit_daylat=params_inst%crit_dayl_at_high_lat-params_inst%crit_dayl_lat_slope* \
+                    (critical_offset_high_lat-abs(grc%latdeg(g)))
+        if (crit_daylat < crit_dayl) then
+           crit_daylat = crit_dayl !maintain previous offset from White 2001 as minimum
+        end if
+    ! Critical day length is constant
+    case(critical_daylight_constant)
+        crit_daylat = crit_dayl
+    case default
+        call endrun(msg="ERROR SeasonalCriticalDaylength crtical_daylight_method not implemented "//errmsg(sourcefile, __LINE__))
+    end select
+
+  end function SeasonalCriticalDaylength
+
+  !-----------------------------------------------------------------------
+  function SeasonalDecidOnset( onset_gdd, onset_gddflag, soilt, soila10, t_a5min, dayl, &
+                               snow_5day, ws_flag, crit_onset_gdd, season_decid_temperate ) &
+                       result( do_onset )
+
+    !
+    ! !DESCRIPTION:
+    ! Function to determine if seasonal decicious leaf onset should happen.
+    !
+    ! !USES:
+    use shr_const_mod   , only: SHR_CONST_TKFRZ
+    ! !ARGUMENTS:
+    real(r8), intent(INOUT) :: onset_gdd      ! onset growing degree days 
+    real(r8), intent(INOUT) :: onset_gddflag  ! Onset freeze flag
+    real(r8), intent(IN)    :: soilt          ! Soil temperture at specific level for this evaluation
+    real(r8), intent(IN)    :: soila10        ! 10-day running mean of the 12cm soil layer temperature (K)
+    real(r8), intent(IN)    :: t_a5min        ! 5-day running mean of min 2-m temperature
+    real(r8), intent(IN)    :: dayl           ! Day length
+    real(r8), intent(IN)    :: snow_5day      ! 5-day average of snow
+    real(r8), intent(IN)    :: ws_flag        ! winter-summer solstice flag (0 or 1)
+    real(r8), intent(IN)    :: crit_onset_gdd ! critical onset growing degree-day sum
+    real(r8), intent(IN)    :: season_decid_temperate  ! If this is a temperate seasonal decidious type 
+    logical :: do_onset                       ! Flag if onset should happen (return value)
+    !
+    ! !LOCAL VARIABLES:
+    real(r8), parameter :: snow5d_thresh_for_onset        = 0.1_r8          ! 5-day snow depth threshold for onset
```

Comment:
Is 39300/2 correct?  This is about 5.4583 hours....

Reply:
or just add a comment about where this value is coming from

Reply:
Yes, 39300/2 is what we are currently using and what we are testing with. You are correct that this is a bit screwy, but changing this could possibly change answers, so I want to leave it alone, unless we do more testing with it changed. So I suppose I can add a comment regarding that.

Reply:
I did add some comments regarding this. So I think this is good.

Reply:
is 39300 used elsewhere in the code as a parameter for a day length trigger for offset?  If so, can we refer to the name of that parameter in the comment so that people could track down where this number came from.

Also, is this the parameter that Leah added to prevent unrealistically early onset in some grids / climates?  If so, let's include that detail on this comment too.

Reply:
Yes, let's include a comment (as I did above) that states: "this is equal to crit_dayl/2, and is used to prevent unrealistically early onset" 

I see you have more on line 1157, but I think a bit is warranted here where the parameter is defined.

Reply:
OK I added more. I'm a little reluctant to refer to the origin of the number, since it likely doesn't mater what the specific value is. I think you could change it within some range and answers will be identical. As such the specific value doesn't matter, but I'm also reluctant to change the value because maybe there is sensitivity to it for a particular point. We'd have to do more testing to see if that's the case, and it's not really worth it to do it.